packages/@aws-cdk/aws-apprunner-alpha/lib/auto-scaling-configuration.ts

```diff
@@ -0,0 +1,198 @@
+import { Construct } from 'constructs';
+import { CfnAutoScalingConfiguration } from 'aws-cdk-lib/aws-apprunner';
+
+/**
+ * Properties of the App Runner Auto Scaling Configuration.
+ */
+export interface AutoScalingConfigurationProps {
+  /**
+   * The name for the Auto Scaling Configuration.
+   *
+   * @default - a name generated by CloudFormation
+   */
+  readonly autoScalingConfigurationName?: string;
+
+  /**
+   * The maximum number of concurrent requests that an instance processes.
+   * If the number of concurrent requests exceeds this limit, App Runner scales the service up.
+   *
+   * Must be between 1 and 200.
+   *
+   * @default 100
+   */
+  readonly maxConcurrency?: number;
+
+  /**
+   * The maximum number of instances that a service scales up to.
+   * At most maxSize instances actively serve traffic for your service.
+   *
+   * Must be between 1 and 25.
+   *
+   * @default 25
+   */
+  readonly maxSize?: number;
+
+  /**
+   * The minimum number of instances that App Runner provisions for a service.
+   * The service always has at least minSize provisioned instances.
+   *
+   *
+   * Must be between 1 and 25.
+   *
+   * @default 1
+   */
+  readonly minSize?: number;
+}
+
+/**
+ * Attributes for the App Runner Auto Scaling Configuration.
+ */
+export interface AutoScalingConfigurationAttributes {
+  /**
+   * The name of the Auto Scaling Configuration.
+   */
+  readonly autoScalingConfigurationName: string;
+
+  /**
+   * The ARN of the Auto Scaling Configuration.
+   */
+  readonly autoScalingConfigurationArn: string;
+
+  /**
+   * The revision of the Auto Scaling Configuration.
+   */
+  readonly autoScalingConfigurationRevision: number;
+}
+
+/**
+ * Represents the App Runner Auto Scaling Configuration.
+ */
+export interface IAutoScalingConfiguration extends cdk.IResource {
+  /**
+   * The ARN of the Auto Scaling Configuration.
+   * @attribute
+   */
+  readonly autoScalingConfigurationArn: string;
+
+  /**
+   * The Name of the Auto Scaling Configuration.
+   * @attribute
+   */
+  readonly autoScalingConfigurationName: string;
+
+  /**
+   * The revision of the Auto Scaling Configuration.
+   * @attribute
+   */
+  readonly autoScalingConfigurationRevision: number;
+}
+
+/**
+ * The App Runner Auto Scaling Configuration.
+ *
+ * @resource AWS::AppRunner::AutoScalingConfiguration
+ */
+export class AutoScalingConfiguration extends cdk.Resource implements IAutoScalingConfiguration {
+  /**
+   * Imports an App Runner Auto Scaling Configuration from attributes
+   */
+  public static fromAutoScalingConfigurationAttributes(scope: Construct, id: string,
+    attrs: AutoScalingConfigurationAttributes): IAutoScalingConfiguration {
+    const autoScalingConfigurationArn = attrs.autoScalingConfigurationArn;
+    const autoScalingConfigurationName = attrs.autoScalingConfigurationName;
+    const autoScalingConfigurationRevision = attrs.autoScalingConfigurationRevision;
+
+    class Import extends cdk.Resource {
```

Comment:
```suggestion
    class Import extends cdk.Resource implements  IAutoScalingConfiguration  {
```

Reply:
In AWS CDK L2s. If the resource name is `Foo`, the construct class of `Foo` should have at least one of fromFooName(), fromFooAttributes() or fromFooArn() and those methods should return `IFoo`. To make it easier, we have a convention to define a tiny `Import` class in those fromXxxx methods that implements IFoo and so we just need to `return new Import()` from those fromXxxx construct methods.

Reply:
Thank you for the review. I understand that while the `fromArn` method has been implemented, additional `fromXXX` methods are needed. After rechecking the design guidelines, it seems better to have a `fromAttributes` method, so I will add that.​​​​​​​​​​​​​​​​

https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#from-attributes